src/System.Private.Xml/src/System/Xml/Dom/XmlName.cs

```diff
@@ -182,18 +183,8 @@ public static int GetHashCode(string name)
             int hashCode = 0;
             if (name != null)
             {
-                unchecked
-                {
-                    for (int i = name.Length - 1; i >= 0; i--)
-                    {
-                        char ch = name[i];
-                        if (ch == ':') break;
-                        hashCode += (hashCode << 7) ^ ch;
-                    }
-                    hashCode -= hashCode >> 17;
-                    hashCode -= hashCode >> 11;
-                    hashCode -= hashCode >> 5;
-                }
+                ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(name.AsSpan(name.LastIndexOf(':') + 1));
+                return Marvin.ComputeHash32(bytes, Marvin.DefaultSeed);
```

Comment:
Supposedely those two functions don't exchange hash code? They are in different assemblies which means they have two copies of Marvin with possibly different seed